packages/wifi_basic/lib/src/extensions.dart

```diff
@@ -1,22 +1,42 @@
 import 'package:wifi_basic/wifi_basic.dart';
 
-extension ToEnumExtension on int? {
-  WiFiGenerations toWifiGeneration() {
-    if (this == null || this! < 0 || this! > 6) {
-      return WiFiGenerations.unknown;
+extension WiFiGenerationsExtension on WiFiGenerations {
+  static WiFiGenerations fromInt(int? value) {
+    switch (value) {
+      case 0:
+      case 1:
+      case 2:
+      case 3:
+        return WiFiGenerations.legacy;
+      case 4:
+        return WiFiGenerations.wifi4;
+      case 5:
+        return WiFiGenerations.wifi5;
+      case 6:
+        return WiFiGenerations.wifi6;
+      default:
+        return WiFiGenerations.unknown;
     }
-    // legacy - if less than 3
-    if (this! <= 3) return WiFiGenerations.legacy;
-    // convert generationInt -> generationEnum
-    return WiFiGenerations.values[this! - 2];
   }
+}
 
-  WiFiNetworkSecurity toWifiNetworkSecurity() {
-    if (this == null ||
-        this! < 0 ||
-        this! > WiFiNetworkSecurity.values.length) {
-      return WiFiNetworkSecurity.unknown;
+extension WiFiNetworkSecurityExtension on WiFiNetworkSecurity {
+  static WiFiNetworkSecurity fromInt(int? value) {
+    switch (value) {
+      case 0:
```

Comment:
case 0 - if for `none` - unknown is just out of range or null. accordng to older logic

Reply:
Just to make sure: I should only remove the `case 0`, right?